packages/api/utils/requestValidation.ts

```diff
@@ -29,3 +33,98 @@ export function shapeValidationForPostRequest<T>(
+ * @returns an ValidationErrorResponse (by throwing an error to the middleware) if there are Clin validation errors
+ */
+
+export function businessRulesValidationForFundingStep(fs: FundingStep): ValidationErrorResponse | undefined {
+  const errors: Array<ClinValidationError> = validateFundingStepClins(fs);
+  if (errors.length) {
+    return createBusinessRulesValidationErrorResponse({ input_validation_errors: errors });
+  }
+  console.log("Business rule validation succeded");
+  return undefined;
+}
+
+export function validateFundingStepClins(fs: FundingStep): Array<ClinValidationError> {
+  return fs.task_orders
+    .flatMap((taskOrder) => taskOrder.clins)
+    .map(validateClin)
+    .reduce((accumulator, validationErrors) => accumulator.concat(validationErrors), []);
+}
+/**
+ * Validates the given clin object
+ * @param clin an object that looks like a Clin
+ * @returns a collection of clin validation errors
+ */
+
+export function validateClin(clin: Clin): Array<ClinValidationError> {
+  const errors = Array<ClinValidationError>();
+  if (new Date(clin.pop_start_date) >= new Date(clin.pop_end_date)) {
+    const obj = {
+      clinNumber: clin.clin_number,
+      validationMessage: ValidationMessage.START_BEFORE_END,
+    };
+    errors.push({
+      ...obj,
+      invalidParameterName: "pop_start_date",
+      invalidParameterValue: clin.pop_start_date,
+    });
+    errors.push({
+      ...obj,
+      invalidParameterName: "pop_end_date",
+      invalidParameterValue: clin.pop_end_date,
+    });
+  }
+  if (new Date() >= new Date(clin.pop_end_date)) {
+    errors.push({
+      clinNumber: clin.clin_number,
+      invalidParameterName: "pop_end_date",
+      invalidParameterValue: clin.pop_end_date,
+      validationMessage: ValidationMessage.END_FUTURE,
+    });
+  }
+
+  if (clin.obligated_funds > clin.total_clin_value) {
+    const obj = {
+      clinNumber: clin.clin_number,
+      validationMessage: ValidationMessage.TOTAL_GT_OBLIGATED,
+    };
+    errors.push({
+      ...obj,
+      invalidParameterName: "obligated_funds",
+      invalidParameterValue: clin.obligated_funds.toString(),
+    });
+    errors.push({
+      ...obj,
+      invalidParameterName: "total_clin_value",
+      invalidParameterValue: clin.total_clin_value.toString(),
+    });
+  }
+  return errors;
+}
+
+export function createBusinessRulesValidationErrorResponse(
+  invalidProperties: Record<string, unknown>
+): ValidationErrorResponse {
+  if (Object.keys(invalidProperties).length === 0) {
+    throw Error("Parameter 'invalidProperties' must not be empty");
+  }
+  Object.keys(invalidProperties).forEach((key) => {
+    if (!key) throw Error("Parameter 'invalidProperties' must not have empty string as key");
+  });
+  // return new ValidationErrorResponse("Invalid input", invalidProperties);
+  const error = createError(400, "Business rules validation failed");
+  error.details = invalidProperties;
+  throw error;
```

Comment:
Can we do:
```suggestion
  throw createError(400, "Business rules validation failed", { details: invalidProperties });
```

Looks like `createError` takes a `properties` argument at the end?